compiler/dio-hdf5/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+nnas_find_package(HDF5 COMPONENTS STATIC QUIET)
+
+if(NOT HDF5_FOUND)
+  message(STATUS "Build dio_hdf5: FAILED (missing HDF5)")
+  return()
+endif(NOT HDF5_FOUND)
+
+file(GLOB_RECURSE SOURCES "src/*.cpp")
+file(GLOB_RECURSE TESTS "src/*.test.cpp")
+list(REMOVE_ITEM SOURCES ${TESTS})
+
+add_library(dio_hdf5 SHARED ${SOURCES})
+target_include_directories(dio_hdf5 PUBLIC include)
+target_include_directories(dio_hdf5 PRIVATE ${HDF5_INCLUDE_DIRS})
```

Comment:
As `HDF5Importer.h` has `#include <H5Cpp.h>` it would be better to use `PUBLIC`

Reply:
I'm not sure but IIUC, the `PRIVATE` keyword applies to the module that link `dio-hdf5` library. So, if we don't want to make that module use `HDF5_INCLUDE_DIRS`, it might as well set it to `PRIVATE`.

Reply:
As I know, when we include `HDF5Importer.h` with `PRIVATE` then user module will not see `${HDF5_INCLUDE_DIRS}` thus compile will fail. If this is not the case then we can use `PRIVATE` or we should use `PUBLIC`

Reply:
Whether it is PUBLIC or PRIVATE, `HDF5Importer.h` can use `H5Cpp.h`.

I set this as `PRIVATE`, because I wanted to expose only `HDF5Importer` to other modules. If we make dio-hdf5 `PUBLIC`, other modules will be possible to access H5Cpp.h when dio-hdf5 is linked.

Reply:
Found out with @jinevening that `H5Cpp.h` is installed in `overlay` folder and other libs provide this. If we make a small project without others then compile fails with `H5Cpp.h` not found...

Reply:
In offline talk with @seanshpark , we've confirmed that @seanshpark is right.

`HDF5Importer.h` is PUBLIC, and it uses `H5Cpp.h`, so we have to set PUBLIC for HDF5 libraries.

What makes things confusing was that HDF5 library with PRIVATE keyword is also compilable. That's because `overlay` (where HDF5 is installed) directory is included by another module (such as luci_import - mio-circle). Technically, that should not be compiled.

Thanks for lesson! @seanshpark